src/main_context.rs

```diff
@@ -26,7 +26,7 @@ impl MainContext {
         }
     }
 
-    pub fn find_source_by_id(&self, source_id: SourceId) -> Option<Source> {
+    pub fn find_source_by_id(&self, source_id: &SourceId) -> Option<Source> {
```

Comment:
@sdroege Strange why find want consume non-copy type

Reply:
That's very weird because I assume that we can get the `SourceId` from the `Source`. However, if no `Source` matches the `SourceId`, you just don't have the `SourceId` anymore... Maybe it is the purpose?

Reply:
Should be by reference, yes